File: shared/src/androidMain/kotlin/com/oddinstitute/crossplatformsvgparser/SvgParser+TagEnded.kt
package com.oddinstitute.crossplatformsvgparser

import com.oddinstitute.crossplatformsvgparser.svg_tags.StyleTag

fun SvgParser.tagEnded()
{
    when (curTagName)
    {
        "style" ->
        {
            val styleTag = StyleTag(curTagText)
            this.styles = styleTag.decodeStyle()
        }

        // when we end the group, it should become inactive
        "g" -> currentGroups.removeLast() //  activeGroup = null

        "defs" -> definitionState = false
    }

    curTagName = null
}

File: iosApp/iosApp/SvgParser+TagEnded.swift
import shared
import Foundation

extension SvgReader
{
    func tagEnded()
    {
        switch (data.curTagName)
        {
        case "style":
            let styleTag = StyleTag(styleText: data.curTagText)
            self.data.styles = styleTag.decodeStyle()
        case "g":
            data.currentGroups.removeLastObject() //  activeGroup = null
        case "defs":
            data.definitionState = false
        default:
            print("weird tag ended: \(data.curTagName ?? "non existent tag name")")
        }
        
        data.curTagName = nil
    }
}
